src/android/test.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * test module
+ *
+ * Created by Appcelerator, Inc
+ * Copyright (c) 2016 Appcelerator. All rights reserved.
+ */
+import { View } from './view';
+
+export default class Test {
+
+	constructor (app, module) {
+		this.app = app;
+		this.module = module;
+	}
+
+	startup () {
+		this.app.logger.info(this.module.id + ' startup');
+	}
+
+	shutdown () {
+		this.app.logger.info(this.module.id + ' shutdown');
+	}
+
+	static createView (opts = {}) {
```

Comment:
The object instantiation pattern is well known and the creator functions can be easily shimmed.

One consideration is the API docs. I can see these changing quite a bit. Perhaps we use this opportunity to find a solution that offers a better experience for our users, specifically searching and offline support.